hs_core/views/resource_folder_hierarchy.py

```diff
@@ -454,3 +459,194 @@ def data_store_file_or_folder_move_or_rename(request, res_id=None):
+    pk = str(pk).strip()
+    try:
+        resource, _, user = authorize(request, pk,
+                                      needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+    except NotFound:
+        return HttpResponse('Bad request - resource not found', status=status.HTTP_400_BAD_REQUEST)
+    except PermissionDenied:
+        return HttpResponse('Permission denied', status=status.HTTP_401_UNAUTHORIZED)
+
+    tgt_path = resolve_request(request).get('target_path', None)
+    src_paths = resolve_request(request).get('source_paths', None)
+    if src_paths is None or tgt_path is None:
+        return HttpResponse('Bad request - src_paths or tgt_path is not included',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    tgt_path = str(tgt_path).strip()
+    if not tgt_path:
+        return HttpResponse('Bad request - tgt_path cannot be empty',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if not tgt_path.startswith('data/contents/'):
+        return HttpResponse('Bad request - tgt_path must start with data/contents/',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if tgt_path.find('/../') >= 0 or tgt_path.endswith('/..'):
+        return HttpResponse('Bad request - tgt_path cannot contain /../',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    istorage = resource.get_irods_storage()
+    tgt_abspath = os.path.join(resource.root_path, tgt_path)
+
+    if not is_folder(istorage, tgt_abspath):
+        return HttpResponse('Bad request - tgt_path is not an existing folder',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    src_paths = json.loads(src_paths)
+
+    for src_path in src_paths:
+        src_path = str(src_path).strip()
+
+        if not src_path.startswith('data/contents/'):
+            return HttpResponse('Bad request - src_path must start with data/contents/',
+                                status=status.HTTP_400_BAD_REQUEST)
+
+        if src_path.find('/../') >= 0 or src_path.endswith('/..'):
+            return HttpResponse('Bad request - src_path cannot contain /../',
+                                status=status.HTTP_400_BAD_REQUEST)
+
+    # TODO: validate that move makes sense and that things are not being moved to subdirectories
+    # of themselves.
+
+    for src_path in src_paths:
+        try:
+            move_to_folder(user, pk, src_path, tgt_path)
+        except SessionException as ex:
+            return HttpResponse(ex.stderr, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        except DRF_ValidationError as ex:
+            return HttpResponse(ex.detail, status=status.HTTP_400_BAD_REQUEST)
+
+    return_object = {'target_rel_path': tgt_path}
+
+    return HttpResponse(
+        json.dumps(return_object),
+        content_type='application/json'
+    )
+
+
+@api_view(['POST'])
+def data_store_rename_file_or_folder(request, pk=None):

```

Comment:
Same response; enables calling this without the API. Following pattern for other routines.